src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/BoundPropertyManager.cs

```diff
@@ -0,0 +1,246 @@
+                    if (currentValue != null && !currentValue.Equals(originalValue))
+                    {
+                        context.ViewData[key] = currentValue;
+                    }
+                }
+            }
+        }
+
+        private static IReadOnlyList<PropertyItem> GetProperties(
+            Type type,
+            IReadOnlyList<BoundPropertySource> allowedSources,
+            string tempDataPrefix,
+            bool requireViewDataPropertySetters)
+        {
+            List<PropertyItem> propertyItems = null;
+            var propertyHelpers = PropertyHelper.GetVisibleProperties(type);
+            for (var i = 0; i < propertyHelpers.Length; i++)
+            {
+                var propertyHelper = propertyHelpers[i];
+                var property = propertyHelper.Property;
+
+                var customAttributes = property.GetCustomAttributes(inherit: false);
+                for (var j = 0; j < customAttributes.Length; j++)
+                {
+                    var attribute = customAttributes[j];
+                    string key;
+                    BoundPropertySource lifetimeKind;
+                    if (attribute is ViewDataAttribute viewData && IsAllowed(BoundPropertySource.ViewData))
+                    {
+                        ValidateViewDataProperty(property, requireViewDataPropertySetters);
+
+                        key = viewData.Key ?? property.Name;
+                        lifetimeKind = BoundPropertySource.ViewData;
+                    }
+                    else if (attribute is TempDataAttribute tempData && IsAllowed(BoundPropertySource.TempData))
+                    {
+                        ValidateTempDataProperty(property);
+
+                        key = tempData.Key ?? tempDataPrefix + property.Name;
+                        lifetimeKind = BoundPropertySource.TempData;
+                    }
+                    else
+                    {
+                        continue;
+                    }
+
+                    propertyItems = propertyItems ?? new List<PropertyItem>();
+                    propertyItems.Add(new PropertyItem(propertyHelper, lifetimeKind, key));
+                }
+            }
+
+            return propertyItems;
+
+            bool IsAllowed(BoundPropertySource source)
+            {
+                for (var i = 0; i < allowedSources.Count; i++)
+                {
+                    if (allowedSources[i] == source)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        private static void ValidateViewDataProperty(PropertyInfo property, bool requireViewDataPropertySetters)
+        {
+            var attributeName = nameof(ViewDataAttribute);
+            var propertyTypeName = property.DeclaringType.FullName;
+            if (property.GetMethod == null || !property.GetMethod.IsPublic)
```

Comment:
Interestingly this never gets hit. `PropertyHelper.GetProperties` filters properties without getters. I left it here because it doesn't particularly hurt to have it.

Reply:
Is this where that happens? https://github.com/aspnet/Mvc/pull/7385/files#diff-df35dc42a7e11bb041b71abb55729fccR129
If it really is not adding value, why keep it? Especially since it appears twice (for both the Validate methods.)